pygmt/helpers/utils.py

```diff
@@ -222,46 +229,66 @@ def data_kind(
     --------
     >>> import numpy as np
     >>> import xarray as xr
+    >>> import pandas as pd
     >>> import pathlib
     >>> data_kind(data=None)
-    'vectors'
-    >>> data_kind(data=np.arange(10).reshape((5, 2)))
-    'matrix'
+    'none'
+    >>> data_kind(data=None, required=False)
+    'arg'
+    >>> [data_kind(data=data) for data in (2, 2.0, True, False)]
+    ['arg', 'arg', 'arg', 'arg']
     >>> data_kind(data="my-data-file.txt")
     'file'
     >>> data_kind(data=pathlib.Path("my-data-file.txt"))
     'file'
-    >>> data_kind(data=None, required=False)
-    'arg'
-    >>> data_kind(data=2.0, required=False)
-    'arg'
-    >>> data_kind(data=True, required=False)
-    'arg'
+    >>> data_kind(data=["data1.txt", "data2.txt"])
+    'file'
     >>> data_kind(data=xr.DataArray(np.random.rand(4, 3)))
     'grid'
     >>> data_kind(data=xr.DataArray(np.random.rand(3, 4, 5)))
     'image'
+    >>> data_kind(data=np.arange(10).reshape((5, 2)))
+    'matrix'
+    >>> data_kind(data=pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]}))
+    'vectors'
+    >>> data_kind(data={"x": [1, 2], "y": [3, 4]})
+    'vectors'
+    >>> data_kind(data=[[1, 2], [3, 4]])
+    'vectors'
+    >>> data_kind(data=[1, 2, 3])
+    'vectors'
     """
-    kind: Literal["arg", "file", "geojson", "grid", "image", "matrix", "vectors"]
+    # data is None and is required.
+    if data is None and required:
+        return "none"
+
+    # A file or a sequence of files
     if isinstance(data, str | pathlib.PurePath) or (
         isinstance(data, list | tuple)
         and all(isinstance(_file, str | pathlib.PurePath) for _file in data)
     ):
-        # One or more files
-        kind = "file"
-    elif isinstance(data, bool | int | float) or (data is None and not required):
-        kind = "arg"
-    elif isinstance(data, xr.DataArray):
-        kind = "image" if len(data.dims) == 3 else "grid"
-    elif hasattr(data, "__geo_interface__"):
-        # geo-like Python object that implements ``__geo_interface__``
-        # (geopandas.GeoDataFrame or shapely.geometry)
-        kind = "geojson"
-    elif data is not None:
-        kind = "matrix"
-    else:
-        kind = "vectors"
-    return kind
+        return "file"
+
+    # An option argument, mainly for dealing with optional virtual files
+    if isinstance(data, bool | int | float) or (data is None and not required):
+        return "arg"
+
+    # A xr.DataArray grid or image
+    if isinstance(data, xr.DataArray):
+        return "image" if len(data.dims) == 3 else "grid"
+
+    # Geo-like Python object that implements ``__geo_interface__`` (e.g.,
+    # geopandas.GeoDataFrame or shapely.geometry)
+    # Reference: https://gist.github.com/sgillies/2217756
+    if hasattr(data, "__geo_interface__"):
+        return "geojson"
+
+    # A 2-D numpy.ndarray
+    if isinstance(data, np.ndarray) and data.ndim == 2:
```

Comment:
We probably can change this line to:
```suggestion
    if hasattr(data, "__array_interface__") and len(data.__array_interface__.shape) == 2:
```
to support more array-like objects that implements the `__array_interface__` protocol (https://numpy.org/doc/stable/reference/arrays.interface.html#object.__array_interface__), but it can be done in a future PR.